lib/ansible/vars/manager.py

```diff
@@ -620,13 +619,16 @@ def set_host_facts(self, host, facts):
         if not isinstance(facts, dict):
             raise AnsibleAssertionError("the type of 'facts' to set for host_facts should be a dict but is a %s" % type(facts))
 
-        if host.name not in self._fact_cache:
-            self._fact_cache[host.name] = facts
-        else:
+        try:
             try:
+                # this is a cache plugin, not a dictionary
+                self._fact_cache.update({host.name, facts})
```

Comment:
I think you want a `:` here instead of a `,`, otherwise you are creating a `set`

Reply:
This PR is a :+1: to me once this is fixed.